### DISCUSSION TITLE
How can we make the feed more social?

### DISCUSSION THREAD

**Discussion by DennyDaVjncj** [community/community#14023] [Other Feature Feedback, Questions, & Ideas]
It would be great if we could easily react to the activity of people we're following. Once that social feature gets implemented, Github will never be the same, and for the better!

**Comment by GoaGill** [comment]
2

**Comment by Hobadee** [comment]
How about no?  I don't come to GitHub to socialize.  You want to Socialize?  There are plenty of platforms for that.  GitHub is a tool to get stuff done, and this "social feed" bullshit is already getting in my way of doing that.

**Reply by DennyDaVjncj** [reply]
I think it's awesome to be able to quickly analyze a good friend's recent push and offer feedback... publicly. On this platform our conversation would be centered on software engineering, coding and devops, etc. I don't know about you, but I love this stuff and would love to chat with programmers from all over the globe!

**Reply by Hobadee** [reply]
> I [...] would love to chat with programmers from all over the globe!

That's what IRC is for.  Or the countless other ways of chatting with people around the globe.  I could throw a chat app on copiers so you can chat with everyone around the globe making copies.  That doesn't mean I should.  GitHub is a tool to do a job.  Just because I can tack "chat" onto something, doesn't mean I should.

**Reply by DennyDaVjncj** [reply]
1. What is the IRC?
2. Is life not an experience worth savoring?! If a sociable feature harms a person's productivity than that person should self-asses, then push some UPDATES! 😆🖖🏾

**Reply by danisztls** [reply]
> I think it's awesome to be able to quickly analyze a good friend's recent push and offer feedback... publicly. On this platform our conversation would be centered on software engineering, coding and devops, etc. I don't know about you, but I love this stuff and would love to chat with programmers from all over the globe!

I just want to have the option. I don't want to have someone else algorithm deciding what I should see and how should I expend my time. Black box social discovery is a cancer and makes no sense for GitHub team to replicate this bad pattern as they don't need to promote engagement to maximize ad revenue.

**Comment by danisztls** [comment]
How can we make the feed LESS social?

I'm receiving updates from people I never interacted with just because they were contributors to a repository I contributed code or opened an issue year ago.

**Reply by DennyDaVjncj** [reply]
That's because you clicked to "watch" that repository; an assumption, but I feel strongly about it. You may be following it too, once you undo either of those possibilities, problem solved

**Comment by twojstaryzdomu** [comment]
If you want social, then there are appropriate services to answer your need. This isn't a site for the socially-deprived.

**Reply by danisztls** [reply]
I agree there are appropriate services for that and I don't want to see those features here though this is a site for everyone regardless of fullness of social needs.

**Comment by MichaelWheldon** [comment]
Not to be rude and I mean this about me too, but no one should care what you’re doing around the edges of github. Knowing what people you’ve not gone out of your way to follow are commenting on or starring or forking !== making a connection or creating a community, quite the opposite.

This kind of wishy washy way of "connecting" people to the "content" on github is a bit sad and is _pretty obviously_ an idea purely to make the platform more marketable and translatable for investors and none coders. It offers no value to the wider coding community.

**Reply by MichaelWheldon** [reply]
The content in these feeds are generated from you working, so they’re basically trying to passively profit from your daily work... dumb as hell that anyone would want this.

**Reply by sujaldev** [reply]
hypothetically if a more social feed were to be desirable what'd be wrong with the content being generated from one's work. It's a win win situation, GitHub gets more profitable (possibly increasing free features and no one cares if it doesn't) all while causing zero impact on the user.

**Reply by twojstaryzdomu** [reply]
Some users find the 'more social' avenue of github as a step towards distracting content, so no, I don't think you can claim it has zero user impact. Users have every right to speak against social features, since this is not a social network.

**Reply by sujaldev** [reply]
how about this as a solution:
[github.com](https://github.com) for getting work done and [social.github.com]() for goofing off. works for everyone!

**Reply by twojstaryzdomu** [reply]
No, it's a shortsighted view, github will be inclined to entice you to their ad-targetting business, based on an algorithmic feed. The roll out of ads is gradual, first you get a feed you didn't ask for, then you get ads in between. All serving as a distraction.

**Reply by sujaldev** [reply]
:disappointed: this is sad

**Reply by twojstaryzdomu** [reply]
If you look at what happened elsewhere where a feed has been added, that's the conclusion to draw. First 'social' content added, then social content interspersed with ads.

**Reply by sujaldev** [reply]
I think GitHub will probably never introduce (intrusive) ads not just because everyone hates them but because of github's demographic, people on github are developers if they don't like ads they'll quickly find ways to get rid of them. Stackoverflow is a good example for implementing ads I think, ads that I can barely notice and a promise of reduced ads on higher reputation, seems like a fair deal.
 
Even if github does do this, as long as the ads would stay only on the homepage it wouldn't be so bad as one could always just choose not to open it.

Here's what I am picturing: the ads could advertise other projects on GitHub.
But I don't know how many projects on GitHub would have the money or the need to advertise, so it may or may not be profitable...

**Reply by twojstaryzdomu** [reply]
Never say never, otherwise you're very naive. What happened to other services gobbled up by M$?

> ads that I can barely notice

All ads are DISTRACTIONS, they have no place on a code development platform.

> a promise of reduced ads on higher reputation, seems like a fair deal.

Such promises are always broken, and users get the bad end of the deal. You're very naive, thinking otherwise.

> one could always just choose not to open it.

One could choose not to use github at all. Do you see where you argument is going?

> ads could advertise other projects on GitHub.

Projects that legitimately need advertising are on other platforms. 

Please do not support pushing whether ads or a useless distracting feed that's all about ads in disguise some time in the future.

**Reply by MichaelWheldon** [reply]
> hypothetically if a more social feed were to be desirable what'd be wrong with the content being generated from one's work. It's a win win situation, GitHub gets more profitable (possibly increasing free features and no one cares if it doesn't) all while causing zero impact on the user.

I absolutely do not want anything I do on github being automatically exposed to people I do not know or have not worked with directly. The implications for that are wide ranging, from creating an environment where people start interacting with things just to be exposed, for fashionable reasons or to look like they’re "active in the community". It’s completely meaningless interactions, designed to get you on the social treadmill to make your interactions on github profitable. It’ll be gradual, but it will eventually dilute the value of following users and starring things.

That’s not even mentioning that what you’re asking for already exists in the Following tab... if you need more social interactions on github, then make them meaningful to you and start directly following and interacting with communities on github. You shouldn’t be looking for ways to passively interact with the community, what a waste of your life and productivity.

**Reply by sujaldev** [reply]
> One could choose not to use github at all.

I don't think the homepage of GitHub serves any functional purpose, but other parts of github do so maybe you can't go down this chain.

> people start interacting with things just to be exposed, for fashionable reasons or to look like they’re "active in the community".

Valid point.

> You shouldn’t be looking for ways to passively interact with the community

I'm not. Neither am I looking forward to ads on GitHub.

> I absolutely do not want anything I do on github being automatically exposed to people I do not know

maybe this could be fixed by it being an opt-in feature

But you guys do make valid points, I suppose it would be a bad idea to make github more social.
